core/src/main/java/io/aiven/klaw/service/UsersTeamsControllerService.java

```diff
@@ -1154,7 +1155,9 @@ public List<TeamModelResponse> getSwitchTeams(String userId) {
           teamModel.setTeamId(switchAllowedTeamId);
           teamModel.setTeamname(
               manageDatabase.getTeamNameFromTeamId(tenantId, switchAllowedTeamId));
-          teamModelList.add(teamModel);
+          if (!StringUtils.isEmpty(teamModel.getTeamname())) {
```

Comment:
Just wondering why would teamname be empty

Reply:
I understand the problem, but like to fix it with a migration script which basically looks over all tables (topic requests, acl requests etc) where necessary

What would happen we see a request in an approved state which has requested by a deleted team ?

Seems like we have a bigger problem ?

Reply:
I havent seen it manifest in another fashion but its something I can definitely look into

Reply:
@aindriu-aiven with a migration script, we can loo over all users/switch team ids, and if not existing, delete them